### PR TITLE
Blank lines formatting

### PR DESCRIPTION
**Pull request by PolinaShneider**
**FORMATTER'S IMPROVEMENTS**
1. Add blank line before new section (except for `configuration`)
2. Remove extra blank lines from the document 

**UPD:**
- Extended vscode `FormattingOptions` with `formatBlankLines` property. 
- Moved default formatting settings (`tabSize: 2` and `insertSpaces: true`) to `DEFAULT_FORMATTING_OPTIONS` for convenient reuse
- For back compatibility with existing formatting tests introduced boolean flag `DEFAULT_FORMATTING_OPTIONS(false)`, allowing to choose whether to perform lines formatting or not

format_sections

### CODE REVIEW

**Comment by PolinaShneider** [comment]
Tests in Linux environment aren't passing even if PR has **zero** changes. I think, there is a problem with VSCode or Travis itself. Tests in MacOS environment are OK. Freezing this PR for now.

**Comment by raipc** [comment]
Why is `--silent` parameter specified in `.travis.yml`?

**Comment by PolinaShneider** [comment]
I don't know. Haven't changed anything in travis config. @VeselovAlex do you have any thoughts about it?

File: server/src/test/indents.test.ts [hunk 1]
```diff
@@ -10,15 +10,15 @@ suite("Formatting indents tests: sections and settings", () => {
       `[configuration]
   width-units = 200
   height-units = 200`,
-      [], FormattingOptions.create(2, true),
+      [], DEFAULT_FORMATTING_OPTIONS(false),
```
Comment (annstriganova) [inline server/src/test/indents.test.ts]: Do we need `DEFAULT_FORMATTING_OPTIONS`?

Comment (PolinaShneider) [reply server/src/test/indents.test.ts]: @VeselovAlex said it is convenient not to set each time same parameters. So, I think, yes. Though it looks a bit weird.

Comment (PolinaShneider) [reply server/src/test/indents.test.ts]: However, as long as we don't need to pass a boolean flag to it, `DEFAULT_FORMATTING_OPTIONS` can be a constant, not a function

**Comment by PolinaShneider** [comment]
Closing it because [this](https://github.com/axibase/charts-language-service/pull/12) PR is merged

**Pull request closed without merging**